vck/src/commonMain/kotlin/at/asitplus/wallet/lib/jws/JwsService.kt

```diff
@@ -272,92 +286,102 @@ object DecryptJwe {
  */
-typealias JwkSetRetrieverFunction = suspend (String) -> JsonWebKeySet?
+fun interface JwkSetRetrieverFunction {
+    operator fun invoke(url: String): JsonWebKeySet?
+}
 
 /**
  * Clients get the parsed [JwsSigned] and need to provide a set of keys, which will be used for verification one-by-one.
  */
-typealias PublicJsonWebKeyLookup = suspend (JwsSigned<*>) -> Set<JsonWebKey>?
-
-typealias VerifyJwsSignatureFun = suspend (jwsObject: JwsSigned<*>, publicKey: CryptoPublicKey) -> Boolean
+fun interface PublicJsonWebKeyLookup {
+    suspend operator fun invoke(
+        jwsObject: JwsSigned<*>,
+    ): Set<JsonWebKey>?
+}
 
-object VerifyJwsSignature {
+fun interface VerifyJwsSignatureFun {
     operator fun invoke(
-        verifySignature: VerifySignatureFun = VerifySignature(),
-    ): VerifyJwsSignatureFun = { jwsObject, publicKey ->
-        catching {
-            verifySignature(
-                jwsObject.plainSignatureInput,
-                jwsObject.signature,
-                jwsObject.header.algorithm.algorithm,
-                publicKey,
-            ).getOrThrow()
-        }.fold(
-            onSuccess = { true },
-            onFailure = {
-                Napier.w("No verification from native code", it)
-                false
-            })
-    }
+        jwsObject: JwsSigned<*>,
+        publicKey: CryptoPublicKey,
+    ): Boolean
 }
 
-typealias VerifyJwsSignatureWithKeyFun = suspend (jwsObject: JwsSigned<*>, signer: JsonWebKey) -> Boolean
+class VerifyJwsSignature(
+    val verifySignature: VerifySignatureFun = VerifySignature(),
+) : VerifyJwsSignatureFun {
+    override operator fun invoke(
+        jwsObject: JwsSigned<*>,
+        publicKey: CryptoPublicKey,
+    ) = catching {
+        verifySignature(
+            jwsObject.plainSignatureInput,
+            jwsObject.signature,
+            jwsObject.header.algorithm.algorithm,
+            publicKey,
+        ).getOrThrow()
+    }.fold(
+        onSuccess = { true },
+        onFailure = {
+            Napier.w("No verification from native code", it)
+            false
+        })
+}
 
-object VerifyJwsSignatureWithKey {
+fun interface VerifyJwsSignatureWithKeyFun {
     operator fun invoke(
-        verifyJwsSignature: VerifyJwsSignatureFun = VerifyJwsSignature(),
-    ): VerifyJwsSignatureWithKeyFun = { jwsObject, signer ->
-        signer.toCryptoPublicKey().getOrNull()?.let {
-            verifyJwsSignature(jwsObject, it)
-        } ?: false.also { Napier.w("Could not convert signer to public key: $signer") }
-    }
+        jwsObject: JwsSigned<*>,
+        signer: JsonWebKey,
+    ): Boolean
 }
 
-typealias VerifyJwsSignatureWithCnfFun = suspend (jwsObject: JwsSigned<*>, cnf: ConfirmationClaim) -> Boolean
+class VerifyJwsSignatureWithKey(
+    val verifyJwsSignature: VerifyJwsSignatureFun = VerifyJwsSignature(),
+) : VerifyJwsSignatureWithKeyFun {
+    override operator fun invoke(
```

Comment:
Considering we will be adding remote PQC capabilities to Signum, this should be suspending